### PR TITLE
Enabling the OSConfig Universal NRP to report audit reasons for the Linux Azure Security Baseline checks to be done later

### PR DESCRIPTION
**Pull request by MariusNi**
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [x] All unit tests are passing.
- [x] I have merged the latest `main` branch prior to this PR submission.
- [x] I submitted this PR against the `main` branch.

### CODE REVIEW

**Review by robertschaedler3 — DISMISSED** [review]

File: src/adapters/mc/OsConfigResource.c [hunk 1]
```diff
@@ -312,6 +312,44 @@ static MI_Result GetReportedObjectValueFromDevice(const char* who, MI_Context* c
     return miResult;
 }
 
+static char* GetReasonFromLog(const char* commandTemplate, char separator, const char* keyName, void* log)
```
Comment (robertschaedler3) [inline src/adapters/mc/OsConfigResource.c:315]: Why not build up the reasons "as you go" instead of searching thtough the logs?

Comment (robertschaedler3) [reply src/adapters/mc/OsConfigResource.c:315]: This feels like there will be alot of file reads/writes

**Review by AhmedBM — APPROVED** [review]

File: src/adapters/mc/OsConfig.schema.mof [hunk 2]
```diff
@@ -3,6 +3,16 @@
 #pragma include ("OMI_BaseResource.mof")
 #pragma include ("MSFT_Credential.mof")
 
+[ClassVersion("1.0.0")]
+class ReasonClass
```
[the remark this answers is not in the document]
Comment (MariusNi) [reply src/adapters/mc/OsConfig.schema.mof:7]: This is a MOF class and its name can be anything. Not a resource class so not called that.

File: src/adapters/mc/OsConfigResource.c (hunk 1 above)
Comment (MariusNi) [reply src/adapters/mc/OsConfigResource.c:315]: Thanks Captain! Let's continue (because we talked about this already) offline.

**Pull request merged**